subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -416,7 +416,9 @@ img_mgmt_upload(struct mgmt_ctxt *ctxt)
 	 * request.
 	 */
 	if (img_mgmt_upload_cb != NULL) {
-		rc = img_mgmt_upload_cb(req.off, action.size, img_mgmt_upload_arg);
+		rc = img_mgmt_upload_cb(req.off, action.size,
+					(const struct image_header *)req.img_data,
+					img_mgmt_upload_arg);
```

Comment:
This is not correct because req.img_data will point to header only when `off` is 0.
I think the `void *arg` of `img_mgmt_upload_cb` could be reused here to just pass current image buffer.
We should change the `img_mgmt_upload_cb` as it is broken from the beginning. Registering of the callback via `img_mgmt_set_upload_cb` with assignment to `img_mgmt_upload_arg` is strange as the callback has access to its env and does not have to have anything being passed to it that way.
On the other hand the callack does not know the env it is being executed in so it needs some data from there.
This should probably be changed so that the callback gets info about current offset, size of current chunk, size of image to come and pointer to current data. Probably via some common structure rather than in arguments,